### PR TITLE
fix(lsp): remove unnecessary file load/write when renaming

### PR DESCRIPTION
**Pull request by tomtomjhj**

### CODE REVIEW

File: runtime/lua/vim/lsp/util.lua [hunk 1]
```diff
@@ -705,11 +705,12 @@ function M.rename(old_fname, new_fname, opts)
   end
 
   for _, b in ipairs(oldbufs) do
-    vim.fn.bufload(b)
     -- The there may be pending changes in the buffer
```
Comment (MariaSolOs) [inline runtime/lua/vim/lsp/util.lua:708]: Since we're already changing this code, could we fix the typo in this comment?

**Review by MariaSolOs — COMMENTED** [review]
@tomtomjhj Is there an associated bug? Could you please describe the problem and the fix?

**Comment by tomtomjhj** [comment]
No, this is just a minor efficiency improvement. 

Previously `rename` would unconditionally read the to-be-renamed file from the disk and write it to the disk. This is redundant in some cases
* If the file is not already loaded, it's not attached to lsp client, so nvim doesn't need to care about this file. 
* If the file is loaded but has no change, it doesn't need to be written.

**Comment by MariaSolOs** [comment]
@tomtomjhj got it. Even for improvements like this, make sure to explain the change in the description ;)

**Pull request merged**